frontend/packages/knative-plugin/src/components/add/event-sources/EventSourceSection.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { useFormikContext, FormikValues } from 'formik';
+import * as _ from 'lodash';
+import AppSection from '@console/dev-console/src/components/import/app/AppSection';
+import { FirehoseList } from '@console/dev-console/src/components/import/import-types';
+import CronJobSection from './CronJobSection';
+import SinkBindingSection from './SinkBindingSection';
+import ApiServerSection from './ApiServerSection';
+import KafkaSourceSection from './KafkaSourceSection';
+import YAMLEditorSection from './YAMLEditorSection';
+import { EventSources } from '../import-types';
+import SinkSection from './SinkSection';
+import AdvancedSection from '../AdvancedSection';
+import { isKnownEventSource } from '../../../utils/create-eventsources-utils';
+import ContainerSourceSection from './ContainerSourceSection';
+
+interface EventSourceSectionProps {
+  projects: FirehoseList;
+  namespace: string;
+}
+
+const EventSourceSection: React.FC<EventSourceSectionProps> = ({ projects, namespace }) => {
+  const { values } = useFormikContext<FormikValues>();
+  let EventSource: React.ReactElement;
+  switch (values.type) {
+    case EventSources.CronJobSource:
+      EventSource = <CronJobSection />;
+      break;
+    case EventSources.SinkBinding:
+      EventSource = <SinkBindingSection />;
+      break;
+    case EventSources.ApiServerSource:
+      EventSource = <ApiServerSection />;
+      break;
+    case EventSources.KafkaSource:
+      EventSource = <KafkaSourceSection />;
+      break;
+    case EventSources.ContainerSource:
+      EventSource = <ContainerSourceSection />;
+      break;
+    default:
+      EventSource = values.type && !isKnownEventSource(values.type) ? <YAMLEditorSection /> : null;
```

Comment:
Once you have the above check you can simply say -
 ```suggestion
      EventSource = <YAMLEditorSection />;
```

Reply:
done